src/main/java/org/mule/extension/ftp/internal/operation/BaseFileSystemOperations.java

```diff
@@ -0,0 +1,343 @@
+   *
+   * @param config                  the {@link FileConnectorConfig} on which the operation is being executed
+   * @param fileSystem              a reference to the host {@link FileSystem}
+   * @param path                    the path of the file to be written
+   * @param content                 the content to be written into the file. Defaults to the current {@link Message} payload
+   * @param createParentDirectories whether or not to attempt creating any parent directories which don't exists.
+   * @param lock                    whether or not to lock the file. Defaults to false
+   * @param mode                    a {@link FileWriteMode}. Defaults to {@code OVERWRITE}
+   * @throws IllegalArgumentException if an illegal combination of arguments is supplied
+   */
+  protected void doWrite(FileConnectorConfig config, FileSystem fileSystem, String path, InputStream content,
+                         boolean createParentDirectories, boolean lock, FileWriteMode mode) {
+    if (content == null) {
+      throw new IllegalContentException("Cannot write a null content");
+    }
+
+    validatePath(path, "path");
+    fileSystem.changeToBaseDir();
+
+    fileSystem.write(path, content, mode, lock, createParentDirectories);
+  }
+
+  /**
+   * Copies the file at the {@code sourcePath} into the {@code targetPath}.
+   * <p>
+   * If {@code targetPath} doesn't exist, and neither does its parent, then an attempt will be made to create depending on the
+   * value of the {@code createParentFolder} argument. If such argument is {@false}, then an {@link IllegalArgumentException} will
+   * be thrown.
+   * <p>
+   * If the target file already exists, then it will be overwritten if the {@code overwrite} argument is {@code true}. Otherwise,
+   * {@link IllegalArgumentException} will be thrown.
+   * <p>
+   * As for the {@code sourcePath}, it can either be a file or a directory. If it points to a directory, then it will be copied
+   * recursively.
+   *
+   * @param config                  the config that is parameterizing this operation
+   * @param fileSystem              a reference to the host {@link FileSystem}
+   * @param sourcePath              the path to the file to be copied
+   * @param targetPath              the target directory where the file is going to be copied
+   * @param createParentDirectories whether or not to attempt creating any parent directories which don't exists.
+   * @param overwrite               whether or not overwrite the file if the target destination already exists.
+   * @param renameTo                the new file name, {@code null} if the file doesn't need to be renamed
+   * @throws IllegalArgumentException if an illegal combination of arguments is supplied
+   */
+  protected void doCopy(FileConnectorConfig config, FileSystem fileSystem, String sourcePath,
+                        String targetPath, boolean createParentDirectories, boolean overwrite, String renameTo) {
+    fileSystem.changeToBaseDir();
```

Comment:
Only applicable if the sourcePath and/or the targetPath are relative. Otherwise, it makes no sense to make this operation.

Reply:
this is an existing call and internally checking if it's absolute path then it's not doing anything so should be good here.